source/extensions/filters/http/header_to_metadata/header_to_metadata_filter.cc

```diff
@@ -120,7 +164,7 @@ bool HeaderToMetadataFilter::addMetadata(StructMap& map, const std::string& meta
     return false;
   }
 
-  std::string decodedValue = std::string(value);
+  std::string decodedValue = value;
   if (encode == envoy::extensions::filters::http::header_to_metadata::v3::Config::BASE64) {
     decodedValue = Base64::decodeWithoutPadding(value);
```

Comment:
drop the above and make this:

```
value = Base64::decodeWithoutPadding(value);
```

and just use `value` for the rest of the method.